src/arith/iter_affine_map.cc

```diff
@@ -347,11 +372,24 @@ class IterMapRewriter : public ExprMutator {
   //                                                              IterSplit(k, scale=1)),
   //                                                      extent=9)
   //                                             scale=1))
-  std::unordered_map<IterSumExpr, IterMark, IterSumHash, IterSumEqual> sum_fuse_map_;
+  // Example(2): expr = i*9 + j*2 + k, i in [0, 4) j in [0, 5) k in [0, 2)
+  //          predicate: 1 <= j*2 + k < 9
+  // Then,    flattened form = IterSum(IterSplit(i, scale=9),
+  //                                   IterSplit(j, scale=2),
+  //                                   IterSplit(k, scale=1))
+  //          normal form    = IterSum(IterSplit(i, scale=9),
```

Comment:
I don't think it's a good idea to rewrite `i*9 + j*2 + k` into such a normal form `(i*9 + (j*2 + k - 1) + 1)`.
The reason is that `j*2 + k - 1` has extent 8, it's weird that the scale of i is 9. 
Actually `i*9 + j*2 + k` with predicate `1 <= j*2 + k < 9` doesn't correspond to an iter, since the value of it is not continuous. It's value is `[1, 8] \union [10, 17], ...`.
So it looks to me that we don't allow lower bound constraints on intermediate iters. We can only bound the final iter. In the above case, to bound `i*9 + j*2 + k >= 1` is legal. while `j*2 + k >= 1` is not acceptable.

Reply:
cc @wrongtest

Reply:
What if we change this example to `i*8 + j*2 + k` , `j*2 + k >= 1`?

Reply:
Then it looks good to me.

Reply:
Then I think you might need to add more test cases to test the sanity.
Such as what if we try to fuse several iters all with lower bound constraints.